Cython/Compiler/Optimize.py

```diff
@@ -5048,6 +5048,38 @@ def visit_SingleAssignmentNode(self, node):
             lhs.lhs_of_first_assignment = True
         return node
 
+    def _check_optimize_method_calls(self, node):
+        function = node.function
+        return (node.is_temp and function.type.is_pyobject and self.current_directives.get(
+                "optimize.unpack_method_calls_in_pyinit"
+                if not self.in_loop and self.current_env().is_module_scope
+                else "optimize.unpack_method_calls"))
+
+    def _check_positional_args_for_method_call(self, positional_args):
+        # Do the positional args imply we can substitute a PyMethodCallNode
```

Comment:
1) The comment you provide makes a good method name:
```suggestion
    def _should_use_PyMethodCallNode(self, positional_args):
```
2) This seems more like a helper function than a method.
3) Consider moving it to `PyMethodCallNode` as a static method, e.g. `PyMethodCallNode.can_be_used_for_posargs(posargs)`.